CHANGELOG.md

```diff
@@ -4,6 +4,302 @@ This changelog goes through all the changes that have been made in each release
+  * Add support for ssh commit signing (#17743)
+  * Allow Loading of Diffs that are too large (#17739)
+  * Add copy button to markdown code blocks (#17638)
+  * Add .gitattribute assisted language detection to blame, diff and render (#17590)
+  * Add `PULL_LIMIT` and `PUSH_LIMIT` to cron.update_mirror task (#17568)
+  * Add Reindex buttons to repository settings page (#17494)
+  * Make SSL cipher suite configurable (#17440)
+  * Add groups scope/claim to OIDC/OAuth2 Provider (#17367)
+  * Add simple update checker to Gitea (#17212)
+  * Migrated Repository will show modifications when possible (#17191)
+  * Create pub/priv keypair for federation (#17071)
+  * Make LDAP be able to skip local 2FA (#16954)
+  * Add nodeinfo endpoint for federation purposes (#16953)
+  * Save and view issue/comment content history (#16909)
+  * Use git attributes to determine generated and vendored status for language stats and diffs (#16773)
+  * Add migrate from Codebase (#16768)
+  * Add migration from GitBucket (#16767)
+  * Add OAuth2 introspection endpoint (#16752)
+  * Add proxy settings and support for migration and webhook (#16704)
+  * Add microsoft oauth2 providers (#16544)
+  * Send registration email on user autoregistration (#16523)
+  * Defer Last Commit Info (#16467)
+  * Support unprotected file patterns (#16395)
+  * Add migrate from OneDev (#16356)
+  * Add option to update pull request by `rebase` (#16125)
+  * Add RSS/Atom feed support for user actions (#16002)
+  * Add support for corporate WeChat webhooks (#15910)
+  * Add a simple way to rename branch like gh (#15870)
+  * Add bundle download for repository (#14538)
+  * Add agit flow support in gitea (#14295)
+* API
+  * Add MirrorUpdated field to Repository API type (#18267)
+  * Fix NPE on try to get tag reference via API (#18245)
+  * Adjust Fork API to allow setting a custom repository name (#18066)
+  * Add API to manage repo tranfers (#17963)
+  * Add API to get file commit history (#17652)
+  * Add API to get issue/pull comments and events (timeline) (#17403)
+  * Add API to get/edit wiki (#17278)
+  * Add API for get user org permissions (#17232)
+  * Add HTML urls to notification API (#17178)
+  * Add API to get commit diff/patch (#17095)
+  * Respond with updated notifications in API (#17064)
+  * Add API to fetch git notes (#16649)
+  * Generalize list header for API (#16551)
+  * Add API Token Cache (#16547)
+  * Allow Token API calls be authorized using the reverse-proxy header (#15119)
+* ENHANCEMENTS
+  * Make the height of the editor in Review Box smaller (4 lines as GitHub) (#18319)
+  * Return nicer error if trying to pull from non-existent user (#18288)
+  * Show pull link for agit pull request also (#18235)
+  * Enable partial clone by default (#18195)
+  * Added replay of webhooks (#18191)
+  * Show OAuth callback error message (#18185)
+  * Increase Salt randomness (#18179)
+  * Add MP4 as default allowed attachment type (#18170)
+  * Remove unused route (#18160)
```

Comment:
Refactoring should not be added here imho

Reply:
```suggestion
```